### PR TITLE
2nd draft of workflow

### PR DESCRIPTION
**Pull request by TDDR**
## Issue This PR Addresses
Part Of: #1743  

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x]  **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

This is an attempt to add a workflow that will build and push our microservice image to the GitHub Container Registry. I am unsure of a couple of lines that I will make a comment on. 

I also do not know how I would test locally. Would I not have to test it right in staging in order for the workflow to trigger?

### CODE REVIEW

File: .github/workflows/deploy-feed-discovery-image.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Build and publish a feed-discovery image to ghcr.io
+on:
+  release:
+    types: [published]
+
+  push:
+    branches:
+      - master
```
Comment (manekenpix) [inline .github/workflows/deploy-feed-discovery-image.yaml:8]: Can you try to change this to the name of the branch you're using and check if this works on your fork?

File: .github/workflows/deploy-feed-discovery-image.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+jobs:
+  docker_publish:
+    runs-on: 'ubuntu-20.04'
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # https://github.com/marketplace/actions/push-to-ghcr
+      - name: Build and publish a Docker image for ${{ github.repository }}
+        uses: macbre/push-to-ghcr@master
```
Comment (humphd) [inline .github/workflows/deploy-feed-discovery-image.yaml:19]: I don't think this action is going to work.  They have a bug with uppercase letters in repo names.

Let's use the official Docker one, https://github.com/docker/build-push-action, and we can use https://github.com/docker/login-action to login to ghcr.io

**Review by humphd — CHANGES_REQUESTED** [review]

**Comment by TDDR** [comment]
Not really sure why these two commits are trying to be added again. 
image

I thought the rebase went fine 
image

but the log shows that the new commits got put on top of mine still
image

File: .github/workflows/deploy-feed-discovery-image.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: src\api\feed-discovery
```
Comment (humphd) [inline .github/workflows/deploy-feed-discovery-image.yaml:23]: Prefer Unix style paths `src/api/feed-discovery`

File: .github/workflows/deploy-feed-discovery-image.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    steps:
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push
+        id: docker_build
```
Comment (humphd) [inline .github/workflows/deploy-feed-discovery-image.yaml:20]: I don't think we need an `id`

File: .github/workflows/deploy-feed-discovery-image.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+on:
+  push:
+    branches:
+      - 'issue#1743'
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
```
Comment (humphd) [inline .github/workflows/deploy-feed-discovery-image.yaml:11]: Let's also include `buildx`, which gives us some benefits when building multi-stage builds:

```
        name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1
```

See https://docs.docker.com/buildx/working-with-buildx/ for more info.

File: .github/workflows/deploy-feed-discovery-image.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: src\api\feed-discovery
+          dockerfile: Dockerfile
+          push: true
+          tags: user/app:latest
```
Comment (humphd) [inline .github/workflows/deploy-feed-discovery-image.yaml:26]: Let's tag this slightly different, so we update the `latest` tag for `feed-discovery` to this new image:

```
tags: feed-discovery:latest
```

I also wonder if we should use `feed-discovery:${{github.sha}}` later on.  But we can add that if we need to go back in time to get old versions.

File: .github/workflows/deploy-feed-discovery-image.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+name: build-and-push-feed-discovery
+
+on:
```
Comment (humphd) [inline .github/workflows/deploy-feed-discovery-image.yaml:3]: We could probably add a path section in here to isolate this to changes in `src/api/feed-discovery` only.  cc @menghif, who just did this for our docs workflows.

**Review by humphd — COMMENTED** [review]
I see you closed this, but wanted to give some feedback.  It's looking pretty close.  Nice work.

**Pull request closed without merging**